src/test/compile-fail/catch-struct-name.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(non_camel_case_types)]
+#![allow(dead_code)]
+#![feature(catch_expr)]
+
+struct catch {} //~ ERROR expected identifier, found keyword `catch`
```

Comment:
Can you add a test for local variables? Just want to make sure that this works:

```rust
fn main() {
    let catch = 0;
}
```

It'd also might be good to see if we can get a better error here (i.e., "`catch` cannot be used as a struct name").